monai/handlers/mlflow_handler.py

```diff
@@ -352,3 +402,46 @@ def _default_iteration_log(self, engine: Engine) -> None:
                     for i, param_group in enumerate(cur_optimizer.param_groups)
                 }
                 self._log_metrics(params, step=engine.state.iteration)
+
+    def _default_dataset_log(self, engine: Engine) -> None:
+        """
+        Execute dataset log operation based on MONAI `Workflow.data_loader.dataset` data.
+        Abstract sample names in a dataset and build a Pandas DataFrame from it. To use this
+        function, every sample in the input dataset must have a filename, which can be fetched
+        from the `filename_or_obj` parameter in the `image_meta_dict` of the sample.
+        This function will log a PandasDataset, generated from the Pandas DataFrame, to MLFlow
+        inputs. For more details about PandasDataset, please refer to this link:
+        https://mlflow.org/docs/latest/python_api/mlflow.data.html#mlflow.data.pandas_dataset.PandasDataset
+
+        Please note that it may take a while to record the dataset if it has too many samples.
+
+        Args:
+            engine: Ignite Engine, it can be a trainer, validator or evaluator.
+
+        """
+        dataloader = getattr(engine, "data_loader", None)
+        dataset = getattr(dataloader, "dataset", None) if dataloader else None
+        if not dataset:
+            raise AttributeError("The dataset of the engine is None. Cannot record it by MLFlow.")
+
+        sample_dict: dict[str, list[str]] = {}
+        sample_dict["images"] = []
+        for sample in dataset:
+            if isinstance(sample, dict):
+                image_name = sample["image_meta_dict"]["filename_or_obj"] if "image_meta_dict" in sample else None
```

Comment:
Maybe better to use `MetaTensor` feature instead?
CC @KumoLiu 

Thanks.

Reply:
Thanks, I will take a look at this.